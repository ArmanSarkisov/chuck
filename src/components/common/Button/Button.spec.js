import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Button } from './index';

describe('Button test', () => {
  let component;
  let mockCallback;
  beforeEach(() => {
    mockCallback = jest.fn();
    component = shallow(<Button text="Some text" onClick={mockCallback} />);
  });

  it('should render Button component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render props text', () => {
    expect(component.text()).toBe('Some text');
  });

  it('should be call event listener', () => {
    const btn = component.find('.bg-blue-500');
    expect(mockCallback.mock.calls.length).toBe(0);
    btn.simulate('click');
    expect(mockCallback.mock.calls.length).toBe(1);
  });
});
