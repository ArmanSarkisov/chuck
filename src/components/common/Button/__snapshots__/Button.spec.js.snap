// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test should render Button component 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  onClick={[MockFunction]}
  type="button"
>
  Some text
</button>
`;
